docs/extending_tilt_with_bazel_support.md

```diff
@@ -0,0 +1,289 @@
+
+```python
+def bazel_k8s(target):
+  return local("bazel run %s" % target)
+```
+
+Easy enough, now in the main body of our Tiltfile we can call it like so:
+
+```python
+k8s_yaml(bazel_k8s(":snack-server"))
+```
+
+And if we Tilt up we should see that Tilt calls Bazel to build the YAML, and by extension its image and then Tilt takes that YAML and deploys it to the cluster. We get all the awesomeness of the HUD with three lines of code in a Tiltfile.
+
+### Building Images
+Now for the second component: images. Let's follow the same strategy and figure out the Bazel command that will just build an image and then integrate it in to the Tiltfile.
+
+From reading about rules_k8s I've learned that it uses rules_docker under the hood for building images. Another [quick glance at rules_docker's documentation](https://github.com/bazelbuild/rules_docker#using-with-docker-locally) reveals that `bazel run //:targetname` will load the image in to the configured docker daemon. Perfect, that's all we need!
+
+To integrate this functionality in to the Tiltfile we will make use of just one function: `custom_build`. Custom build takes a command that produces a docker image and tells Tilt about it so it can track it and manage its insertion in to the YAML that we provided previously. Here's what it looks like:
+
+```python
+custom_build(
+  'bazel/snack',
+  'bazel run //snack:image',
+  [],
+  tag="image",
+)
+```
+
+This tells Tilt to run the command (`bazel run //snack:image`) and to assert after it has completed that an image named bazel/snack will exist with the tag "image". If it doesn't then the build fails.
+
+We know what the image will be ahead of time because of Bazel's deterministic nature. Bazel uses a formula for determining the name of the image that it will push: "bazel/" + the directory that the build file is in". This image is tagged with the name of the rule that created it, which in our case is "image".
+
+The third argument, the empty list, is a list of dependencies. We'll fill that in later. For now let's wrap this in a function called `bazel_build` and add it to the main section of our Tiltfile
+
+```python
+k8s_yaml(bazel_k8s(":snack-server"))
+bazel_image('bazel/snack', '//snack:image')
+```
+
+Now Tilt can get YAML from Bazel, get an image from Bazel, insert it into the YAML and track its status in the Kubernetes cluster.
+
+## Dealing with Dependencies
+There's a problem with our Tiltfile as it stands: it only picks up changes if the Tiltfile changes. It doesn't understand what dependencies go in to generating the YAML _or_ the code dependencies that go in to generating the docker image. Luckily for us Bazel makes it easy to ask for those dependencies and Tilt makes it easy to tell it about those dependencies. Let's start by getting the dependencies for the YAML and then move on to the dependencies for the docker image.
```

Comment:
this paragraph is hard to follow. you need to already be immersed in how Tilt works to understand it.

Maybe better to tell the story from the user's perspective?

Like,

```
If you started Tilt now, it would appear to work! Run `tilt up` and all the images get deployed to your cluster.

Then you make a change to main.go. Nothing gets updates. Why not?

It's not enough for Tilt to know how to build & deploy an image. It needs to know what files should trigger a new build and deploy.

Luckily for us Bazel makes it easy to ask for those files and Tilt makes it easy to tell it about those files.
```

Reply:
Ooh yes I like this.